backend/analysis-engine/src/main/java/org/sonarsource/sonarlint/core/analysis/container/global/ModuleRegistry.java

```diff
@@ -34,37 +33,29 @@
 public class ModuleRegistry {
   private static final SonarLintLogger LOG = SonarLintLogger.get();
 
-  private final ConcurrentHashMap<Object, ModuleContainer> moduleContainersByKey = new ConcurrentHashMap<>();
+  private final ConcurrentHashMap<String, ModuleContainer> moduleContainersByKey = new ConcurrentHashMap<>();
   private final SpringComponentContainer parent;
 
   public ModuleRegistry(SpringComponentContainer parent, Supplier<List<ClientModuleInfo>> modulesProvider) {
     this.parent = parent;
     modulesProvider.get().forEach(this::registerModule);
   }
 
-  public ModuleContainer registerModule(ClientModuleInfo moduleInfo) {
-    return moduleContainersByKey.computeIfAbsent(moduleInfo.key(), id -> createContainer(id, moduleInfo.fileSystem()));
+  public void registerModule(ClientModuleInfo moduleInfo) {
+    moduleContainersByKey.computeIfAbsent(moduleInfo.key(), id -> createContainer(id, moduleInfo.fileSystem()));
   }
 
   private ModuleContainer createContainer(Object moduleKey, @Nullable ClientModuleFileSystem clientFileSystem) {
     LOG.debug("Creating container for module '" + moduleKey + "'");
-    var moduleContainer = new ModuleContainer(parent, false);
+    var moduleContainer = new ModuleContainer(parent);
     if (clientFileSystem != null) {
       moduleContainer.add(clientFileSystem);
     }
     moduleContainer.startComponents();
     return moduleContainer;
   }
 
-  public ModuleContainer createTransientContainer(Iterable<ClientInputFile> filesToAnalyze) {
```

Comment:
Why do we remove notion of the transient container? It is needed for single file analysis for files outside of the config scope. TransientModuleFileSystem.class left unused after this change. We need to remove it as well if it's intentional change.

Reply:
> It is needed for single file analysis for files outside of the config scope

This is already not supported anymore, all files are part of a config scope. For the LS for example, they create an "artificial" root configuration scope that serves for files outside folders.

Good catch for the TransientModuleFileSystem, removing it now